docs/src/proposals/comprehensive-compute-fees.md

```diff
@@ -0,0 +1,190 @@
+
+Over 4 1/2 hours of Mainnet beta monitoring with a 200k instruction cap:
+675kPX9MHTjS2zt1qfr1NYHuzeLXfQM9H24wFSUt1Mp8 (Radium v3?)
+  - exceeded 79669 times
+  - exceeded by between 1 and 800k units
+5fNfvyp5czQVX77yoACa3JJVEhdRaWjPuazuWgjhTqEH (Mango!)
+  - exceeded 60 times
+  - exceeded by < 100k units
+
+Both transactions contain 2 or more instructions and both are running the same program in each instruction
+
+Takeaways:
+- If the transactions were split into single instructions per transaction they would be below the tx-wide compute limit of 200k.
+
+## Mainnet beta program performance measurements
+
+```
+Per-program instruction characteristics over 5 minutes of mainnet
+Program                                                    count               Total us        total compute units     units/us
+                                                                     max  med  avg   sd     max   med    avg     sd  med avg  sd
+
+27haf8L6oxUeXrHrgEgsexjSY5hbVUWEmvv9Nyxg8vQv Radium v2       329    9082 1466 2298 3062  126131  58074  65212 24949   32  39  28
+4MNPdKu9wFMvEeZBMt3Eipfs5ovVWTJb31pEXDJAAxX5                  45     264   82   86   36  789       789    789     0   10  8   22
+675kPX9MHTjS2zt1qfr1NYHuzeLXfQM9H24wFSUt1Mp8 (Radium v3?)   6183   15544  858 1310 1355  179980  43283  55484 31286   53  45  79
+7vxeyaXGLqcp66fFShqUdHxdacp4k4kwUpRSSeoZLCZ4                  27    3411 2101 2232  531  149039 115065 114843 18152   53  50 173
+9KEPoZmtHUrBbhWN1v1KWLMkkvwY6WLtAVUCPRtRjP4z                 483    3136  747  646  445   48069  35015  27749 14983   48  47 129
+9xQeWvG816bUx9EPjHmaT23yvVM2ZWbrrpZb9PusVFin Serum v3     115445   31749  483  550  430   26388   2489   5232  5181    7   7   7
+ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL                 114    1485  651  681  158   47894  27498  27820  3966   42  40 158
+CBuCnLe26faBpcBP2fktp4rp8abpcAnTWft6ZrP5Q4T                  228    1188  121  240  285   35918   8010  11542  9716   63  54  94
+DjVE6JNiYqPL2QXyCUUh8rNjHrbz9hXHNYt99MQ59qw1                  82    2058  813  845   25   34272  28994  29111  1768   36  34 113
+EhhTKczWMGQt46ynNeRX1WfeagwwJd7ufHvCDjRxjo5Q                 634    1557  111  292  294   29485   6100  11707  8519   53  47 101
+FjJ5mhdRWeuaaremiHjeQextaAw1sKWDqr3D7pXjgztv                  86    2484  951  968  226   40565  40565  40565     0   43  42 179
+J21zqcffYQU2NUJrvAKhqpKZLQK1YaB9dY5kmcxkMQvQ                  72     952  332  334   96   10645  10315  10331   128   32  31 108
+SwaPpA9LAaLfeLi3a68M4DjnLqgtticKg6CnyNwgAC8                  203    9947 1076 1198  896   66633  58706  57275  4676   53  47  61
+TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA                72286    8453  101  112   98    4429   3531   3026   683   29  27  31
+Vote111111111111111111111111111111111111111                        39278  121  160  400
+11111111111111111111111111111111                                      24   11   11    3
+```
+
+- Handy compute unit conversions
```

Comment:
I guess this is with the caveat that the min TPS here is a limit per contract/account write lock.

Parallelised execution within a `TransactionBatch` of 128 txs, using work-stealing or other load balancing method is still a meaningful way of hiding the worst-case execution latency and catering to the median, both in cu/us, and median cu overall.

So even assuming a moderate amount of worker threads (16) and targetting the median cu of 5000, let's say, at 10cu/us, one could still achieve an average of 500us per tx at a 4+ms batch time. That's still a moderate 32K TPS. This will only scale with num_cores, although perhaps we may want to increase the batch sizes so that there are enough average time txs to hide the worst case time txs.

Also for this reason, I'd prefer to see average rather than median execution time.

This analysis depends on the average execution time staying low. I think this is fairly reasonable since raydium is already a pretty complex contract that has multiple inner instructions, but seems to have a reasonable median execution time. I think with more contracts/asset pairs, the parallelisability would only increase.